src/renderer/components/+preferences/extension-preference-items.injectable.ts

```diff
@@ -2,49 +2,40 @@
  * Copyright (c) OpenLens Authors. All rights reserved.
  * Licensed under MIT License. See LICENSE in root directory for more information.
  */
-import { pipeline } from "@ogre-tools/fp";
-import { getInjectable, getInjectionToken } from "@ogre-tools/injectable";
-import { filter, overSome } from "lodash/fp";
+import { getInjectable, lifecycleEnum } from "@ogre-tools/injectable";
+import type { IComputedValue } from "mobx";
 import { computed } from "mobx";
-import rendererExtensionsInjectable from "../../../extensions/renderer-extensions.injectable";
-import type { LensRendererExtension } from "../../../extensions/lens-renderer-extension";
-import type { RegisteredAppPreference } from "./app-preferences/app-preference-registration";
 
-interface ExtensionPreferenceItem extends RegisteredAppPreference {
-  extension: LensRendererExtension;
-}
+import rendererExtensionsInjectable from "../../../extensions/renderer-extensions.injectable";
 
-export const extensionPreferenceItemInjectionToken = getInjectionToken<ExtensionPreferenceItem>({
-  id: "extension-preference-item-injection-token",
-});
+import type { RegisteredAppPreference } from "./app-preferences/app-preference-registration";
 
 const extensionsPreferenceItemsInjectable = getInjectable({
   id: "extension-preference-items",
 
-  instantiate: (di) => {
+  instantiate: (di, pathParams: IComputedValue<Record<string, string>>): IComputedValue<RegisteredAppPreference[]> => {
     const extensions = di.inject(rendererExtensionsInjectable);
+    const { extensionId, tabId } = pathParams.get();
+    const extension = extensions.get().find((extension) => extension.sanitizedExtensionId === extensionId);
+
+    if (!extension) {
+      return computed(() => []);
+    }
+  
+    const preferences = extension.appPreferences.map(preference => ({
+      id: preference.id || preference.title,
+      ...preference,
+    }));
 
     return computed(() => {
-      const enabledExtensions = extensions.get();
-
-      return pipeline(
-        di.injectMany(extensionPreferenceItemInjectionToken),
-
-        filter((item) =>
-          overSome([
-            isNonExtensionItem,
-            isEnabledExtensionItemFor(enabledExtensions),
-          ])(item),
-        ),
-      );
+      if (tabId) {
+        return preferences.filter(preference => preference.showInPreferencesTab == tabId);
+      }
+
+      return preferences.filter(preference => !preference.showInPreferencesTab);
     });
   },
+  lifecycle: lifecycleEnum.singleton,
```

Comment:
nit: this is the default

Reply:
Yes, but typescript won't allow to leave injectable without this line:
<img width="573" alt="errors without lifecycle" src="https://user-images.githubusercontent.com/9607060/172667078-035eb433-d9ed-4e0b-8b25-fb89af5f344f.png">

Reply:
Ah I see now, this is because you are using an injection parameter (pathParams). @jansav Should specifying singleton with an injection parameter be even allowed?

Why don't you just use `di.inject(currentPathParametersInjectable)` within this injectable?

Reply:
That is what I thought. I had thought that the typings I did prevented this, but I guess not.

Reply:
Changed lifecycle to `transient`.